### PR TITLE
Add Support for MacOS

### LINKED ISSUE #9: Add support for MacOS
Only Windows and Linux are supported for now. It would be great to support MacOS.

### PR DESCRIPTION
**Pull request by Isard7777**
Close #9

### CODE REVIEW

**Comment by Isard7777** [comment]
Ideally, I'd still like to do the yuvk -> rgba conversion in a separate shader to avoid redundancy in the other shaders, but I think I've already used up the budget.

File: src/videoviewer_internal.cpp [hunk 1]
```diff
@@ -1,633 +1,549 @@
...
+const std::vector<uint16_t> indices = {
+    0, 1, 2, 2, 3, 0
+};
+
+static uint32_t g_window_width = 0;
+static uint32_t g_window_height = 0;
+static void framebuffer_size_callback(GLFWwindow* window, int width, int height);
+
+VideoViewer_Internal::VideoViewer_Internal()
+    : m_compute_shader(std::make_unique<Shader>()),
```
Comment (julienalofs) [inline src/videoviewer_internal.cpp:68]: Shouldn't this variable `m_compute_shader` be renamed?

File: src/videoviewer_internal.cpp [hunk 2]
```diff
@@ -1,633 +1,549 @@
...
+
+using namespace Deltacast;
+struct Vertex
+{
+   glm::vec2 position;
+   glm::vec2 texcoord;
+};
+
+const std::vector<Vertex> vertices = {
+    {{-1.0f, -1.0f},  {0.0f, 0.0f}}, // {triangle vertices x, y} , {texture x, y}
```
Comment (julienalofs) [inline src/videoviewer_internal.cpp:53]: Can't we avoid vertices flip by changing those vertices coordinates?

**Review by julienalofs — CHANGES_REQUESTED** [review]
Since you changed the shaders, did you validate all packing conversion in all supported video resolution?

**Comment by Isard7777** [comment]
> Since you changed the shaders, did you validate all packing conversions in all supported video resolutions?

I have validated my development (including all pixel formats, especially YUV 422 10-bit) on the following resolutions:

720x480
720x576
1280x720
1920x1080
2048x1536
3840x2160
4096x2160
All formats are working as expected, except for YUV 422 10-bit with MSB issues on 720p, 2048x1536, and 4096x2160. Upon investigation, I found that this issue was already present prior to my changes.

I will explore whether specific shaders can be introduced for these formats, as we have done previously in dScope.

File: sample/colorbar.hpp [hunk 3]
```diff
@@ -45,7 +45,7 @@ namespace Deltacast
          reserved_rgb_444_12_be, /*< RGB 4:4:4 12bit big endian without any padding */
          reserved_rgb_444_16_le, /*< RGB 4:4:4 16bit little endian without any padding */
          reserved_rgb_444_16_be, /*< RGB 4:4:4 12bit big endian without any padding */
-         bgr_444_8_le_msb,
+         bgr_444_8_le_msb, /*< BGR 4:4:4 8bit little endian with msb padding (30/32 bits) */
```
Comment (julienalofs) [inline sample/colorbar.hpp:48]: 24/32

File: src/videoviewer_internal.cpp [hunk 4]
```diff
@@ -306,15 +306,15 @@ bool VideoViewer_Internal::release()
...
+   m_conversion_shader.reset();
    m_render_shader.reset();
 
    return true;
 }
 
 void VideoViewer_Internal::create_shaders(const char* compute_shader_name)
 {
-   m_compute_shader->compile(std::string(vertex_shader_flip_vertices), std::string(compute_shader_name));
+   m_conversion_shader->compile(std::string(vertex_shader_flip_vertices), std::string(compute_shader_name));
```
Comment (julienalofs) [inline src/videoviewer_internal.cpp]: compute_shader_name aussi à renommer

File: sample/main.cpp [hunk 5]
```diff
@@ -121,12 +121,12 @@ int main(int argc, char** argv)
    std::mutex synchronisation_mutex;
    Deltacast::VideoViewer viewer;
 
-   int texture_width = 1920;
-   int texture_height = 1080;
+   int texture_width = 720;
```
Comment (julienalofs) [inline sample/main.cpp]: C'est volontaire d'avoir rentré ces modifications dans le sample ?

**Comment by Isard7777** [comment]
> > Since you changed the shaders, did you validate all packing conversions in all supported video resolutions?
> 
> I have validated my development (including all pixel formats, especially YUV 422 10-bit) on the following resolutions:
> 
> 720x480 720x576 1280x720 1920x1080 2048x1536 3840x2160 4096x2160 All formats are working as expected, except for YUV 422 10-bit with MSB issues on 720p, 2048x1536, and 4096x2160. Upon investigation, I found that this issue was already present prior to my changes.
> 
> I will explore whether specific shaders can be introduced for these formats, as we have done previously in dScope.

Since I wasn't able to implement this quickly enough, I created an issue to address the problem later, in accordance with our discussion.

**Review by julienalofs — APPROVED** [review]

**Pull request merged**